starlite/asgi.py

```diff
@@ -49,17 +62,20 @@ def _traverse_route_map(self, path: str, scope: "Scope") -> Tuple[RouteMapNode,
         path_params: List[str] = []
         current_node = self.app.route_map
         components = ["/", *[component for component in path.split("/") if component]]
-        for component in components:
+        for idx, component in enumerate(components):
             components_set = cast("ComponentsSet", current_node["_components"])
             if component in components_set:
                 current_node = cast("RouteMapNode", current_node[component])
                 if "_static_path" in current_node:
                     self._handle_static_path(scope=scope, node=current_node)
                     break
                 continue
-            if PathParamPlaceholder in components_set:
+            if PathParamNode in components_set:
+                current_node = cast("RouteMapNode", current_node[PathParamNode])
+                if PathParameterTypePathDesignator in components_set:
+                    path_params.append(normalize_path(path.removeprefix("/".join(path.split("/")[:idx]))))
```

Comment:
testing with 

```py
@get("/{path:path}")
def hello_world(path: Path) -> dict[str, Any]:
    """Hi."""
    return {"hello": str(path)}
```
Requesting `http://localhost:8000/a/b/c/d/e` returns `{"hello":"/a/b/c/d/e"}` but perhaps the leading `/` should not be included in the path, as it is outside the parameter.

I found a test from both flask and starlette that shows they both exclude the leading `/`:

```python
def test_static_folder_with_ending_slash():
    app = flask.Flask(__name__, static_folder="static/")

    @app.route("/<path:path>")
    def catch_all(path):
        return path

    rv = app.test_client().get("/catch/all")
    assert rv.data == b"catch/all"
```
and

```python
    # Test path conversion
    response = client.get("/path/some/example")
    assert response.status_code == 200
    assert response.json() == {"path": "some/example"}
    assert (
        app.url_path_for("path-convertor", param="some/example") == "/path/some/example"
    )
```

Reply:
ok, i modified it. please test and see if this works for you now.

Reply:
Sorry dude, have shutdown and gone to bed so can't test rn but looks like it would've done the trick👍